include/pthread.h

```diff
@@ -0,0 +1,376 @@
+#ifndef _PTHREAD_H
+#define _PTHREAD_H
+
+/* FIXME: This should probably live somewhere else but Zephyr doesn't
```

Comment:
Might be a better idea to just define it in the minimal libc instead, even if nothing there will use it.